mlir/include/mlir/Dialect/AMDGPU/IR/AMDGPU.td

```diff
@@ -433,6 +433,47 @@ def AMDGPU_LDSBarrierOp : AMDGPU_Op<"lds_barrier"> {
   let assemblyFormat = "attr-dict";
 }
 
+def AMDGPU_SchedBarrierOpOpt : I32EnumAttr<"sched_barrier_opt_enum",
```

Comment:
I think we can use `I32BitEnumAttrCase` here? With that we can avoid the array in the below and it also have better direct assembly/parse/print support via the infra.